SharkCage/CageServiceInstaller/CageServiceInstaller.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections;
+using System.ComponentModel;
+using System.Configuration.Install;
+
+namespace CageServiceInstaller
+{
+    [RunInstaller(true)]
+    public partial class CageServiceInstaller : Installer
+    {
+        const string service_name = "shark-cage-service";
+
+        public override void Install(System.Collections.IDictionary state_saver)
+        {
+            base.Install(state_saver);
+        }
+
+        public override void Commit(System.Collections.IDictionary saved_state)
+        {
+            base.Commit(saved_state);
+
+            string target_dir = Context.Parameters["DP_TargetDir"].ToString();
+            InstallAndStartService(target_dir);
+        }
+
+        public override void Rollback(IDictionary saved_state)
+        {
+            base.Rollback(saved_state);
+            UninstallService();
+        }
+
+        public override void Uninstall(System.Collections.IDictionary saved_state)
+        {
+            base.Uninstall(saved_state);
+            UninstallService();
```

Comment:
Could it be changing the order of uninstalling is as easy as switching these two lines? (So the installer won't show the running service message)

Reply:
Haha had the same idea but unfortunately not, the complete functions is called after popping up the running service message (and pressing continue...).